### PR TITLE
Add custom timestamp filtration

### PR DESCRIPTION
**Pull request by zigaLuksic**
Closes https://github.com/sentinel-hub/eo-learn/issues/498

@ColinMoldenhauer is this sufficient for your needs? I would also like to add you as a [co-author](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) (since the idea and solution suggestion are yours), but I'd need an e-mail under which to add you.

### CODE REVIEW

File: io/eolearn/tests/test_sentinelhub_process.py [hunk 1]
```diff
@@ -524,6 +524,28 @@ def test_get_available_timestamps_with_missing_data_collection_service_url(self)
         assert len(timestamps) == 4
         assert all(timestamp.tzinfo is not None for timestamp in timestamps)
 
+    def test_get_available_timestamps_custom_filtration(self):
+        """Chekcs that the custom filtration works as intended."""
```
Comment (ColinMoldenhauer) [inline io/eolearn/tests/test_sentinelhub_process.py:528]: Typo: Checks

**Review by ColinMoldenhauer — COMMENTED** [review]
Looks good to me, thank you for the quick action!

**Comment by ColinMoldenhauer** [comment]
> Closes #498
> 
> @ColinMoldenhauer is this sufficient for your needs? I would also like to add you as a [co-author](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) (since the idea and solution suggestion are yours), but I'd need an e-mail under which to add you.

As an e-mail, you can use colin.moldenhauer [at] posteo.de, thank you!

**Comment by jgersak** [comment]
Code looks good. 
I would just add test where we would check sth. like: 
```python
len(set(timestamp)) == len(timestamp filtration) != len(timestamp))
```
and make doc strings unified whit existing doc strings.

**Review by jgersak — APPROVED** [review]

**Pull request merged**